iree/compiler/Dialect/HAL/Conversion/HALToVM/ConvertAllocatorOps.cpp

```diff
@@ -45,8 +45,19 @@ class AllocatorAllocateConstOpConversion
     auto constName = (parentFuncOp.getName() + "_const_" +
                       std::to_string(allocateUniqueId(parentFuncOp)))
                          .str();
+
+    // VM supports a minimum of 8-bit elements. Round 1-bit bools to 8-bits.
+    auto value = op.value().cast<ElementsAttr>();
+    if (value.getType().getElementType().isInteger(1)) {
```

Comment:
I'm not that familiar with the complete list of steps in IREE's conversion pipelines. Is there an earlier place where a cast from bool to i8 would be sufficient? By the time we're here in the HAL dialect (HALToVM), I'd expect us to be missing some higher level context that would help us understand where such a conversion makes the most sense. This could benefit from an expanded comment here, or a section within a more holistic design doc (similar to https://google.github.io/iree/design-docs/codegen-passes).

Reply:
Yeah this is still too low-level - the HAL cannot support 1 bit data types either.